internal/datastore/mongo_store.go

```diff
@@ -2,89 +2,131 @@ package datastore
 	"github.com/SimplQ/simplQ-golang/internal/models"
 
+	"go.mongodb.org/mongo-driver/bson"
 	"go.mongodb.org/mongo-driver/bson/primitive"
 	"go.mongodb.org/mongo-driver/mongo"
 	"go.mongodb.org/mongo-driver/mongo/options"
 )
 
 // A Structure with Collections frequently used and a pointer to the client
 type MongoDB struct {
-    Client *mongo.Client
-    Queue *mongo.Collection
-    Token *mongo.Collection
+	Client *mongo.Client
+	Queue  *mongo.Collection
+	Token  *mongo.Collection
 }
 
 func NewMongoDB() *MongoDB {
-    // Use local development mongodb instance if env variable not set
-    uri := "mongodb://root:example@localhost:27017/?maxPoolSize=20&w=majority"
-    
-    if val, ok := os.LookupEnv("MONGO_URI"); ok {
-        uri = val
+	// Use local development mongodb instance if env variable not set
+	uri := "mongodb://root:example@localhost:27017/?maxPoolSize=20&w=majority"
+
+	if val, ok := os.LookupEnv("MONGO_URI"); ok {
+		uri = val
 	}
 
-    log.Println("Connecting to MongoDB...")
+	log.Println("Connecting to MongoDB...")
+
+	client, err := mongo.Connect(context.TODO(), options.Client().ApplyURI(uri))
 
-    client, err := mongo.Connect(context.TODO(), options.Client().ApplyURI(uri))
-    
-    if err != nil {
-        log.Fatal(err)
-    }
+	if err != nil {
+		log.Fatal(err)
+	}
 
-    new_mongodb := MongoDB {
-        client,
-        client.Database("simplq").Collection("queue"),
-        client.Database("simplq").Collection("token"),
-    }
+	new_mongodb := MongoDB{
+		client,
+		client.Database("simplq").Collection("queue"),
+		client.Database("simplq").Collection("token"),
+	}
 
-    log.Println("Successfully connected to MongoDB!")
+	log.Println("Successfully connected to MongoDB!")
 
-    return &new_mongodb
+	return &new_mongodb
 }
 
 func (mongodb MongoDB) CreateQueue(queue models.Queue) models.QueueId {
-    // Set id to empty so its generated by mongoDB
-    queue.Id = ""
+	// Set id to empty so its generated by mongoDB
+	queue.Id = ""
 
-    result, err := mongodb.Queue.InsertOne(context.TODO(), queue)
+	result, err := mongodb.Queue.InsertOne(context.TODO(), queue)
 
-    if err != nil {
-        log.Fatal(err)
-    }
- 
-    stringId := result.InsertedID.(primitive.ObjectID).Hex()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	stringId := result.InsertedID.(primitive.ObjectID).Hex()
 
-    return models.QueueId(stringId)
+	return models.QueueId(stringId)
 }
 
-func (mongodb MongoDB) ReadQueue(models.QueueId) models.Queue {
-    panic("Not implemented")
+func (mongodb MongoDB) ReadQueue(id models.QueueId) models.Queue {
+	result := models.Queue{}
+
+	// convert id string to ObjectId
+	queueId, _ := primitive.ObjectIDFromHex(string(id))
+
+	err := mongodb.Queue.FindOne(context.TODO(), bson.M{"_id": queueId}).Decode(&result)
+
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	return result
 }
 
-func (mongodb MongoDB) PauseQueue(models.QueueId) {
-    panic("Not implemented")
+func (mongodb MongoDB) PauseQueue(id models.QueueId) {
```

Comment:
The code for this function and resume function is identical.

Let's have one `setIsPaused(id models.QueueId, isPaused boolean)` private function and then call it with true/false in `PauseQueue/ResumeQueue`

Reply:
Added setIsPaused()